### ISSUE TITLE
Need to check playback duration

### ISSUE THREAD

**Issue by plbossart** [thesofproject/sof-test#353]
maybe I am missing something but I don't see a check that e.g. a 2s file is played in 2s.
I saw a case where playback was happening at 2x speed because of a borked firmware, and nothing was reported.

**Comment by xiulipan** [comment]
@plbossart Is it OK we try some test case like
`time aplay -Dhw:0,0 -f dat /dev/zero -d 2`
And measure the measured time with the played time

**Comment by plbossart** [comment]
> @plbossart Is it OK we try some test case like
> `time aplay -Dhw:0,0 -f dat /dev/zero -d 2`
> And measure the measured time with the played time

yes, exactly. You may want to use 10s so that the initial setup (which can take 200-300ms) does not impact the results too much, and place an error threshold at 5% of the expected duration.

**Comment by marc-hb** [comment]
The `timeout` command can help, example: https://github.com/thesofproject/sof-test/pull/961

See also:
- https://github.com/thesofproject/sof-test/issues/489

**Comment by marc-hb** [comment]
Duplicate of?
- #256
